zapret configs test script added

FlowersOfMind opened the pull request:
Добавил скрипт для автоматического тестирования стратегий.
Запуск 'run test zapret.bat'.
В targets также можно добавить свои сервисы для проверки.
Сохранение результатов в csv.
<img width="577" height="505" alt="image" src="https://github.com/user-attachments/assets/1a2cfc4d-7dcc-427b-8733-ce437c37e4f8" />

FlowersOfMind committed:
zapret configs test script added

Flowseal commented:
Смысла от такой проверки - около нуля. Для дискорда нужно другие домены тоже смотреть (гейтвей, апдейтс), при этом разные tls версии. Для ютуба нужно смотреть на гуглвидео и т.п.

FlowersOfMind committed:
Update targets.txt with new endpoints

Added various endpoints for Discord, YouTube, Google, Cloudflare, and public DNS.

FlowersOfMind committed:
Add testing for different TLS versions and parallel execution

FlowersOfMind commented:
Не работает из-за кириллицы, проблемы с кодировкой, поправлю

FlowersOfMind committed:
Пересохранение файла test zapret.ps1 в кодировке UTF-8 with BOM без изменения содержимого

FlowersOfMind committed:
UTF-8 with BOM encoding

FlowersOfMind committed:
Encoding issues fixed.

FlowersOfMind committed:
add interactive mode to select specific configs

FlowersOfMind commented:
Добавил ручной выбор тестов

otakypyjie33-hub commented:
Перезапустил, точно так же в конце теста окно командной строки просто автоматически закрывается...Может быть из за того, что я разворачиваю его на весь экран не знаю (Так удобней результаты сравнивать) по факту я даже результаты последнего теста посмотреть не успеваю оно просто мгновенно закрывается и ни следа..

FlowersOfMind committed:
keep test window open after script completion

otakypyjie33-hub commented:
> > Перезапустил, точно так же в конце теста окно командной строки просто автоматически закрывается...Может быть из за того, что я разворачиваю его на весь экран не знаю (Так удобней результаты сравнивать) по факту я даже результаты последнего теста посмотреть не успеваю оно просто мгновенно закрывается и ни следа..
> 
> Попробуйте сейчас

Cейчас не могу, проверю позже только хочу уточнить...Мне заново копировать содержимое [Files changed 3] или именно то, что в [keep test window open after script completion] ?
П.с было бы конечно удобнее если бы его можно было уже скачать просто готовыми файлами...Там в архиве, а не "Собирать" самому...

FlowersOfMind committed:
add zapret service check

FlowersOfMind commented:
Добавил проверку сервиса, если установлен, то тест не запустится

FlowersOfMind committed:
load targets before mode selection

FlowersOfMind committed:
remove leftover call to deleted Wait-ForExit

FlowersOfMind committed:
Remove TLS 1.0 and TLS 1.1

FlowersOfMind committed:
Improve output formatting

FlowersOfMind committed:
Improve compatibility

FlowersOfMind committed:
Improve compatibility 2

FlowersOfMind committed:
Improve compatibility 3

FlowersOfMind committed:
Handle Write-Host failures on legacy consoles

FlowersOfMind committed:
Redo compatibility improvements using a simpler approach

FlowersOfMind committed:
require any key to exit PowerShell

V3nilla commented:
> под это сделать можешь? https://hyperion-cs.github.io/dpi-checkers/ru/tcp-16-20/

Тогда автору надо реализовать, чтобы скрипт считал не только ответ от сервера, но и обьем данных поступающего с него, дабы детектить 16кб блок и выводить юзеру, что его секут и пора менять стратегии

FlowersOfMind committed:
added dpi checkers and increased test speed

V3nilla commented:
> Добавил dpi checkers, а еще использую другой подход для параллельных тестов от чего скорость тестирования повысилась в разы 🤗 <img alt="image" width="953" height="1026" src="https://private-user-images.githubusercontent.com/210693535/525244750-7685440e-9175-43c1-aa91-6e48d4fed915.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU0Mzg5MDUsIm5iZiI6MTc2NTQzODYwNSwicGF0aCI6Ii8yMTA2OTM1MzUvNTI1MjQ0NzUwLTc2ODU0NDBlLTkxNzUtNDNjMS1hYTkxLTZlNDhkNGZlZDkxNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxMVQwNzM2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTcxNWJiYTNlYjNlMTYwMGJhZjZhNTFkN2VjM2FhMTRkZDQzMzczMmU1YzdkMWRiZGI1MzNmYmU0NzM2M2VkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DiQQxt1Nrq2TJNQrAFpJgYkNG6jC7dpwXhzhYG3yERc">

Предлагаю еще добавить - Тест на триггер со стороны ТСПУ, то есть - даже коннекта до сервера нету, ни одного байта не дошло до сервера. В основном блокируется "UDP 443", из-за чего не происходит коннект, когда тот-же TCP 443 или ICMP - нет, для создания видимости, что всё ок

FlowersOfMind committed:
backup Ipset list

FlowersOfMind commented:
> Проблема с DPI чекером в том, что он зависит от установленного айписета пользователя, о чём сам пользователь не знает. Нужно либо предупреждать пользователя, либо автоматически менять айписет (но если пользователь аварийно завершит выполнение скрипта, то возможно останется не с тем айписетом, с которым обычно сидел).

Добавил логику для автоматической смены ip(возвращается обратно в случае окончания, ошибки, остановки) и если было аварийное закрытие(через X, или выключился компьютер), то сохраняется backup предыдущих настроек и при следующем запуске настройки восстанавливаются
+ добавил уведомление и предупреждение о об этом

FlowersOfMind commented:
> А батник вообще не запускается, закрывается и открывается, поведение как у трояна какого-то ))

привет, это потому что это и есть троян, я заразил твой комп вирусами (￣y▽￣)╭ Ohohoho.....



> Спасибо за тестер и парочка хотелок 1 Если возможна краткая аналитика - сколько тригеров и каких какая стратегия задела в конце. 2. После окончания теста сохранение результата рядом в текстовик. 3. После окончанию теста возврат в меню выбора тестов, что заново не запускать скрипт.

привет, посмотрю что можно добавить как свободное время выдастся

FlowersOfMind committed:
added analytics and save results

FlowersOfMind committed:
moved tests to service

Goga-Rid commented:
@FlowersOfMind 

### Без прав админа не выполняется ps скрипт, пытался намутить с помощью ярлыка толку ноль, ибо обращение идет именно к скрипту

<img width="532" height="560" alt="image" src="https://github.com/user-attachments/assets/e4235fb6-99a5-489e-9a50-364ea25dcf5e" />

Dronatar commented:
> @FlowersOfMind
> 
> ### Без прав админа не выполняется ps скрипт, пытался намутить с помощью ярлыка толку ноль, ибо обращение идет именно к скрипту
> <img alt="image" width="532" height="560" src="https://private-user-images.githubusercontent.com/112984775/530040072-e4235fb6-99a5-489e-9a50-364ea25dcf5e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY1OTI2MTcsIm5iZiI6MTc2NjU5MjMxNywicGF0aCI6Ii8xMTI5ODQ3NzUvNTMwMDQwMDcyLWU0MjM1ZmI2LTk5YTUtNDg5ZS05YTUwLTM2NGVhMjVkY2Y1ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNFQxNjA1MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjRlZGZjM2YyNjViNmYwZmM4YTg4OWFkZjQzMjRiYTk3MmRiZWExYTc0MWFmNjM2NGUxODVmMjg4ZDQwYzI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bHQm8kKxLBgJyDI6_KeQBzyRrcY6xvIjqUQV4SOoHtM">

А у меня вообще никак не запускалось, не админа/от админа/PowerShell, не-а, пришлось делать .bat с
```
@echo off
setlocal

powershell -NoProfile -ExecutionPolicy Bypass -Command "Start-Process -FilePath powershell.exe -ArgumentList '-NoProfile', '-ExecutionPolicy', 'Bypass', '-Command', '& {& ''тут путь к .ps1''}'; exit $LASTEXITCODE"

exit %errorlevel%
```
И уже потом запускать от админа.
Хотя и тут, он не смог найти winws. Странно как-то.

FlowersOfMind commented:
> > > @FlowersOfMind
> > > ### Без прав админа не выполняется ps скрипт, пытался намутить с помощью ярлыка толку ноль, ибо обращение идет именно к скрипту
> > > <img alt="image" width="532" height="560" src="https://private-user-images.githubusercontent.com/112984775/530040072-e4235fb6-99a5-489e-9a50-364ea25dcf5e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY1OTI2MTcsIm5iZiI6MTc2NjU5MjMxNywicGF0aCI6Ii8xMTI5ODQ3NzUvNTMwMDQwMDcyLWU0MjM1ZmI2LTk5YTUtNDg5ZS05YTUwLTM2NGVhMjVkY2Y1ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNFQxNjA1MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjRlZGZjM2YyNjViNmYwZmM4YTg4OWFkZjQzMjRiYTk3MmRiZWExYTc0MWFmNjM2NGUxODVmMjg4ZDQwYzI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bHQm8kKxLBgJyDI6_KeQBzyRrcY6xvIjqUQV4SOoHtM">
> > 
> > 
> > А у меня вообще никак не запускалось, не админа/от админа/PowerShell, не-а, пришлось делать .bat с
> > ```
> > @echo off
> > setlocal
> > 
> > powershell -NoProfile -ExecutionPolicy Bypass -Command "Start-Process -FilePath powershell.exe -ArgumentList '-NoProfile', '-ExecutionPolicy', 'Bypass', '-Command', '& {& ''тут путь к .ps1''}'; exit $LASTEXITCODE"
> > 
> > exit %errorlevel%
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > И уже потом запускать от админа. Хотя и тут, он не смог найти winws. Странно как-то.
> 
> [#3423 (comment)](https://github.com/Flowseal/zapret-discord-youtube/discussions/3423#discussioncomment-15321740)
> 
> #### Тут у чела вроде получилось реализовать, он тоже использовал батник, хз...

Через service.bat запустить пробовали?

Goga-Rid commented:
> > > > > > @FlowersOfMind
> 
> > > 
> 
> > > 
> 
> > > > > > ### Без прав админа не выполняется ps скрипт, пытался намутить с помощью ярлыка толку ноль, ибо обращение идет именно к скрипту
> 
> > > 
> 
> > > 
> 
> > > > > > <img alt="image" width="532" height="560" src="https://private-user-images.githubusercontent.com/112984775/530040072-e4235fb6-99a5-489e-9a50-364ea25dcf5e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY1OTI2MTcsIm5iZiI6MTc2NjU5MjMxNywicGF0aCI6Ii8xMTI5ODQ3NzUvNTMwMDQwMDcyLWU0MjM1ZmI2LTk5YTUtNDg5ZS05YTUwLTM2NGVhMjVkY2Y1ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNFQxNjA1MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjRlZGZjM2YyNjViNmYwZmM4YTg4OWFkZjQzMjRiYTk3MmRiZWExYTc0MWFmNjM2NGUxODVmMjg4ZDQwYzI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bHQm8kKxLBgJyDI6_KeQBzyRrcY6xvIjqUQV4SOoHtM">
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > А у меня вообще никак не запускалось, не админа/от админа/PowerShell, не-а, пришлось делать .bat с
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > @echo off
> 
> > > 
> 
> > > 
> 
> > > > > setlocal
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > powershell -NoProfile -ExecutionPolicy Bypass -Command "Start-Process -FilePath powershell.exe -ArgumentList '-NoProfile', '-ExecutionPolicy', 'Bypass', '-Command', '& {& ''тут путь к .ps1''}'; exit $LASTEXITCODE"
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > exit %errorlevel%
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > И уже потом запускать от админа. Хотя и тут, он не смог найти winws. Странно как-то.
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > [#3423 (comment)](https://github.com/Flowseal/zapret-discord-youtube/discussions/3423#discussioncomment-15321740)
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > #### Тут у чела вроде получилось реализовать, он тоже использовал батник, хз...
> 
> > > 
> 
> > > 
> 
> > > Через service.bat запустить пробовали?
> 
> > 
> 
> > #### Да, появляется окно и сразу пропадает, результаты тестов не появляются в директории test results
> 
> 
> 
> Скачай отсюда https://github.com/FlowersOfMind/zapret-discord-youtube/tree/test-zapret и попробуй запустить, проблема была в том, что окно закрывалось при ошибке (например если сервис установлен)

Точно я забыл про это, надо в README прописать будет что нужно отключать либо варнинг закинуть при активации через service.bat

ekungurov commented:
Тест, я так понял, curl'ом делается. Кто-нибудь знает, виндовый curl уважает настройки системных настроек DNS-over-HTTPS ?

sgurov commented:
Здравствуйте!

Спасибо за тесты, они мне помогли определить 5 рабочих стратегий: ALT10, FAKE TLS AUTO, FAKE TLS AUTO ALT3, SIMPLE FAKE, SIMPLE FAKE ALT2.

Затем запустил DPI checkers тесты, получил такой результат:

=== ANALYTICS ===
general (FAKE TLS AUTO).bat : OK: 50, FAIL: 22, UNSUP: 0, BLOCKED: 0
general (SIMPLE FAKE ALT2).bat : OK: 54, FAIL: 18, UNSUP: 0, BLOCKED: 0
general (FAKE TLS AUTO ALT3).bat : OK: 35, FAIL: 37, UNSUP: 0, BLOCKED: 0
general (ALT10).bat : OK: 39, FAIL: 33, UNSUP: 0, BLOCKED: 0
general (SIMPLE FAKE).bat : OK: 38, FAIL: 34, UNSUP: 0, BLOCKED: 0
True
True
False
True
False

Best config: general (SIMPLE FAKE).bat

Подскажите, почему лучшей стратегией определилась general (SIMPLE FAKE).bat?

Если проанализировать результаты, то кажется, что general (SIMPLE FAKE ALT2).bat имеет наилучшее соотношение ok/fail. Но может я что-то упускаю, что не отображено в результатах?

delta2404 commented:
Здраствуйте!

Что делать если интернет конфликтует с программой Zapret?
Программа сама по себе не работает.

![photo_2026-01-02_18-35-25](https://github.com/user-attachments/assets/d7ee4e11-0e84-4a09-9251-0f13899d13c3)

harmonicano commented:
> Здраствуйте!
> 
> Что делать если интернет конфликтует с программой Zapret? Программа сама по себе не работает.

создай issue [тут](https://github.com/Flowseal/zapret-discord-youtube/issues/new?template=bug_report.yml). не мусорь в пуллах.
